src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -105,11 +107,53 @@ class ComService(
     val enrichedProbationSearchResults = entityProbationSearchResult.mapNotNull {
       val licences =
         licenceRepository.findAllByCrnAndStatusCodeIn(it.identifiers.crn, LicenceStatus.IN_FLIGHT_LICENCES)
-      val prisonerSearchResult = this.prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(it.identifiers.noms))
 
-      // If an empty list has been returned, there are no relevant licences relating to search for the offender
+      /**
+       *  If an empty list has been returned, we do not have a licence for the person currently but still need to return
+       *  a result for the PP as they may want to create a licence for them
+       */
       if (licences.isEmpty()) {
-        null
+        val isNomisNumberPopulated = !it.identifiers.noms.isNullOrBlank()
+
+        if (isNomisNumberPopulated) {
+          val prisonerSearchResults = this.prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(it.identifiers.noms))
```

Comment:
we probably don't want to do a separate call for each search result.. would be better to batch them 🤔

Reply:
So you've currently got: 
```
val records = ...

for r in records {
    val licences = ...
    if (licence.isEmpty) {
         createNewRecord(r)
    } else {
         createExistingRecord(r)
    }
}
```

Reply:
I'd possibly be tempted to 

```
val recordsWithLicences: List<Pair<Record, Licence?>> = records.map { it, getLicence(it) }
```
where getLicence, gets the relevant licence if one exists

Then:
```
val prisonersForUnstartedLicences: List<Prisoner search results> = ...findPrisoners(recordsWithLicences.filter { it.second == null }) 
``` 

Then:
```
recordsWithLicences.map { (record, licence) ->
   it (licence == null)  
      createUnstartedRecord(record, prisonersForUnstartedLicences.find { it.prisonNumber = record.prisonNumber})
  else
      createUnstartedRecord(record, prisonersForUnstartedLicences.find { it.prisonNumber = record.prisonNumber})
}
```